src/gui/tracks/TrackContentWidget.cpp

```diff
@@ -82,16 +95,27 @@ TrackContentWidget::TrackContentWidget( TrackView * parent ) :
 
 
 
-
-
-
 void TrackContentWidget::updateBackground()
 {
+	// use snapSize to determine number of lines to draw
+	float snapSize = getGUI()->songEditor()->m_editor->getSnapSize();
```

Comment:
I'm getting a crash here when I start the application and immediately close it.

Reply:
Could you describe this issue some more? For some reason, it works fine on my end.

Reply:
The steps to reproduce are:
1. Build a debug build of the application.
2. Start the application in a debugger.
3. Close the application.

Before the application is fully quit the debugger breaks at the location linked above. The stack trace is:
```
lmms::gui::TrackContentWidget::updateBackground(lmms::gui::TrackContentWidget * const this) (/home/lmms-build/LMMS/lmms-others/src/gui/tracks/TrackContentWidget.cpp:101)
lmms::gui::TrackContentWidget::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void ** _a) (/home/lmms-build/LMMS/lmms-others/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_TrackContentWidget.cpp:115)
libQt5Core.so.5![Unknown/Just-In-Time compiled code] (Unbekannte Quelle:0)
lmms::Model::dataChanged(lmms::Model * const this) (/home/lmms-build/LMMS/lmms-others/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_Model.cpp:154)
lmms::AutomatableModel::setValue(lmms::AutomatableModel * const this, const float value) (/home/lmms-build/LMMS/lmms-others/src/core/AutomatableModel.cpp:318)
lmms::AutomatableModel::setRange(lmms::AutomatableModel * const this, const float min, const float max, const float step) (/home/lmms-build/LMMS/lmms-others/src/core/AutomatableModel.cpp:424)
lmms::ComboBoxModel::clear(lmms::ComboBoxModel * const this) (/home/lmms-build/LMMS/lmms-others/src/core/ComboBoxModel.cpp:52)
lmms::ComboBoxModel::~ComboBoxModel(lmms::ComboBoxModel * const this) (/home/lmms-build/LMMS/lmms-others/include/ComboBoxModel.h:52)
lmms::ComboBoxModel::~ComboBoxModel(lmms::ComboBoxModel * const this) (/home/lmms-build/LMMS/lmms-others/include/ComboBoxModel.h:53)
libQt5Core.so.5!QObjectPrivate::deleteChildren() (Unbekannte Quelle:0)
libQt5Widgets.so.5!QWidget::~QWidget() (Unbekannte Quelle:0)
lmms::gui::TrackContainerView::~TrackContainerView(lmms::gui::TrackContainerView * const this) (/home/lmms-build/LMMS/lmms-others/src/gui/editors/TrackContainerView.cpp:127)
lmms::gui::SongEditor::~SongEditor(lmms::gui::SongEditor * const this) (/home/lmms-build/LMMS/lmms-others/include/SongEditor.h:68)
lmms::gui::SongEditor::~SongEditor(lmms::gui::SongEditor * const this) (/home/lmms-build/LMMS/lmms-others/include/SongEditor.h:68)
libQt5Core.so.5!QObjectPrivate::deleteChildren() (Unbekannte Quelle:0)
libQt5Widgets.so.5!QWidget::~QWidget() (Unbekannte Quelle:0)
lmms::gui::Editor::~Editor(lmms::gui::Editor * const this) (/home/lmms-build/LMMS/lmms-others/include/Editor.h:84)
lmms::gui::SongEditorWindow::~SongEditorWindow(lmms::gui::SongEditorWindow * const this) (/home/lmms-build/LMMS/lmms-others/include/SongEditor.h:175)
lmms::gui::SongEditorWindow::~SongEditorWindow(lmms::gui::SongEditorWindow * const this) (/home/lmms-build/LMMS/lmms-others/include/SongEditor.h:175)
lmms::gui::MainWindow::~MainWindow(lmms::gui::MainWindow * const this) (/home/lmms-build/LMMS/lmms-others/src/gui/MainWindow.cpp:256)
lmms::gui::MainWindow::~MainWindow(lmms::gui::MainWindow * const this) (/home/lmms-build/LMMS/lmms-others/src/gui/MainWindow.cpp:259)
libQt5Core.so.5!QObject::event(QEvent*) (Unbekannte Quelle:0)
libQt5Widgets.so.5!QApplicationPrivate::notify_helper(QObject*, QEvent*) (Unbekannte Quelle:0)
libQt5Core.so.5!QCoreApplication::notifyInternal2(QObject*, QEvent*) (Unbekannte Quelle:0)
libQt5Core.so.5!QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) (Unbekannte Quelle:0)
libQt5Core.so.5![Unknown/Just-In-Time compiled code] (Unbekannte Quelle:0)
libglib-2.0.so.0!g_main_context_iteration (Unbekannte Quelle:0)
libQt5Core.so.5!QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (Unbekannte Quelle:0)
libQt5Core.so.5!QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (Unbekannte Quelle:0)
libQt5Core.so.5!QCoreApplication::exec() (Unbekannte Quelle:0)
main(int argc, char ** argv)
```

As you can see the `SongEditor` is in the process of being deleted but is accessed by the code in `updateBackground` which in turn is triggered by `ComboBoxModel::clear`.

The problem might potentially be fixed by checking if the instances that are used in `TrackContentWidget::updateBackground` are not `nullptr`.

However, I wonder if the call to `clear` in the destructor of `ComboBoxModel` is not a problem as well. Why should the destructor manipulate the model and emit a signal if it is about to be destroyed anyway? I am aware that this was not introduced by your PR but it might be worth discussing this with other people to make the code more stable so that we do not get into the situations seen in the stack trace.

Reply:
Ah, I see.

The (quick and dirty) way of fixing it is just to go to `TrackContentWidget.cpp`, check if `songEditor == NULL`, and returning if true.

But I did experiment with the destructor in `ComboBoxModel.h`. For what it's worth, simply commenting out `clear` fixes the issue and doesn't seem to adversely affect the program. I will check for leaks with `valgrind` once I get it to work.

Reply:
Ah, I see.

The (quick and dirty) way of fixing it is just to go to `TrackContentWidget.cpp`, check if `songEditor == NULL`, and returning if true.

But I did experiment with the destructor in `ComboBoxModel.h`. For what it's worth, simply commenting out `clear` fixes the issue and doesn't seem to adversely affect the program (although I wouldn't take that as a reliable indicator of anything).

I'll ask around.

Reply:
Just asked around the Discord, and no one could find a good reason for `clear` being there. So I guess I'll go with the second solution.

Reply:
Looks good. The crash is gone.